src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -155,7 +156,7 @@
      * How big (in megabytes) an on-disk cache to keep of GitHub API responses. Cache is per repo, per credentials.
      */
     private static /*mostly final*/ int cacheSize =
-            Math.min(1024, Math.max(0, Integer.getInteger(GitHubSCMSource.class.getName() + ".cacheSize", 20)));
+            Math.min(1024, Math.max(0, Integer.getInteger(GitHubSCMSource.class.getName() + ".cacheSize", isWindows() ? 0 : 20)));
```

Comment:
🐛 This is fine if you wanted to *just* set it at restart via system property, but if you want to make it settable via system property at runtime, you can't use a static field and will need a getter function that retrieves the cacheSize instead. 

The general logic looks reasonable though, assuming 0 disables the cache (I have not looked to verify).